osu.Game/Screens/OnlinePlay/DrawableRoomPlaylistItem.cs

```diff
@@ -85,27 +90,43 @@ protected override void LoadComplete()
         {
             base.LoadComplete();
 
-            SelectedItem.BindValueChanged(selected => maskingContainer.BorderThickness = selected.NewValue == Model ? 5 : 0, true);
+            SelectedItem.BindValueChanged(selected =>
+            {
+                bool isCurrent = selected.NewValue == Model;
+
+                if (!valid.Value)
+                {
+                    // Don't allow selection when not valid.
```

Comment:
Not sure that this code path is ever reached, `OnClick` already seems to disallow selection preemptively. But I guess it shouldn't hurt to have it just in case either way.